packages/manifold/contracts/physicalclaim/PhysicalClaimCore.sol

```diff
@@ -0,0 +1,337 @@
+        bytes32[] memory merkleProof;
+        uint8 variation;
+        (instanceId, burnItemIndex, merkleProof, variation) = abi.decode(data, (uint256, uint256, bytes32[], uint8));
+
+        PhysicalClaim storage physicalClaimInstance = _getPhysicalClaim(instanceId);
+
+        // A single  can only be sent in directly for a burn if:
+        // 1. There is no cost to the burn (because no payment can be sent with a transfer)
+        // 2. The burn only requires one NFT (one burnSet element and one count)
+        _validateReceivedInput(physicalClaimInstance.burnSet.length, physicalClaimInstance.burnSet[0].requiredCount);
+
+        _getAvailablePhysicalClaimCount(physicalClaimInstance.totalSupply, physicalClaimInstance.redeemedCount, 1);
+
+        // Check that the burn token is valid
+        BurnItem memory burnItem = physicalClaimInstance.burnSet[0].items[burnItemIndex];
+
+        // Can only take in one burn item
+        if (burnItem.tokenSpec != TokenSpec.ERC721) {
+            revert InvalidInput();
+        }
+        PhysicalClaimLib.validateBurnItem(burnItem, msg.sender, id, merkleProof);
+
+        // Do burn and redeem
+        _burn(burnItem, address(this), msg.sender, id, 1, "");
+        _redeem(instanceId, physicalClaimInstance, from, 1, variation, "");
+    }
+
+    function _validateReceivedInput(uint256 length, uint256 requiredCount) private pure {
+        if (length != 1 || requiredCount != 1) {
+            revert InvalidInput();
+        }
+    }
+
+    /**
+     * Send funds to receiver
+     */
+    function _forwardValue(address payable receiver, uint256 amount) private {
+        (bool sent, ) = receiver.call{value: amount}("");
+        if (!sent) {
+            revert TransferFailure();
+        }
+    }
+
+    /**
+     * Burn all listed tokens and check that the burn set is satisfied
+     */
+    function _burnTokens(PhysicalClaim storage burnRedeemInstance, BurnToken[] memory burnTokens, uint256 burnRedeemCount, address owner, bytes memory data) private {
+        // Check that each group in the burn set is satisfied
+        uint256[] memory groupCounts = new uint256[](burnRedeemInstance.burnSet.length);
+
+        for (uint256 i; i < burnTokens.length;) {
+            BurnToken memory burnToken = burnTokens[i];
+            BurnItem memory burnItem = burnRedeemInstance.burnSet[burnToken.groupIndex].items[burnToken.itemIndex];
+
+            PhysicalClaimLib.validateBurnItem(burnItem, burnToken.contractAddress, burnToken.id, burnToken.merkleProof);
+
+            // Always 721s right now, so always 1
+            _burn(burnItem, owner, burnToken.contractAddress, burnToken.id, 1, data);
+            groupCounts[burnToken.groupIndex] += burnRedeemCount;
+
+            unchecked { ++i; }
+        }
+
+        for (uint256 i; i < groupCounts.length;) {
+            if (groupCounts[i] != burnRedeemInstance.burnSet[i].requiredCount * burnRedeemCount) {
+                revert InvalidBurnAmount2();
```

Comment:
@wwhchung I keep hitting this...

Reply:
https://dashboard.tenderly.co/tx/goerli/0x75235da780d9236db6eeac7ec3c0be31205cb4fca0ba2d32eff51cef0f3e87a7

Reply:
Nvm I was giving bad input when burning